src/tabcenter.js

```diff
@@ -99,9 +99,22 @@ TabCenter.prototype = {
     for (let identity of identities) {
       const identityItem = document.createElement("div");
       identityItem.className = "newtab-menu-identity";
-      identityItem.addEventListener("mouseup", () => {
-        this.hideNewTabMenu();
-        browser.tabs.create({ cookieStoreId: identity.cookieStoreId });
+      identityItem.addEventListener("mouseup", async e => {
+        switch (e.which) {
+        case 1:
+          browser.tabs.create({ cookieStoreId: identity.cookieStoreId });
+          this.hideNewTabMenu();
+          break;
+        case 2:
+          var currentTab = await browser.tabs.query({ active: true });
```

Comment:
The currentTab naming would be semantically true if you selected the [0]th element right now